canonical-web-and-design/ubuntu.com
# Advantage shop VAT

## Done

Implemented a VAT/purchase preview endpoint that returns a VAT total (when applicable) and actual total and presents it to the user during the payment modal journey.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage/subscribe
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Whilst logged in, make a product selection, add it to the cart, and click "Buy Now" and QA the following two flows:

#### VAT
- In the payment modal, select United Kingdom (VAT is always applicable)
- After a short wait, VAT and total should appear below the form
- Change the country to United States (VAT is not applicable)
- After a short wait, VAT should disappear and total should update
- Change the country to France (VAT is applicable without a VAT number)
- After a short wait, VAT and total should appear below the form
- Fill out the VAT number field with "FR77784608416" (Credit Agricole's VAT number)
- After a short wait, VAT should disappear and total should update

#### Prorating
- In the payment modal, allow a moment for the subtotal to appear
- If you have an existing subscription that ends in less than a year, the end date of each plan should change to reflect that, and the first plan will contain a note explaining the change.
  - If you don't have such a subscription, ask Scott to set you up with one.

## Issue / Card

Fixes #7647 and #8235
